src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -376,35 +375,44 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       stay() using data.copy(startMode = Option(startMode))
   }
 
-  private def handleCallRetried(callKey: OutputKey,
-                                  retryStatus: ExecutionStatus,
-                                  returnCode: Option[Int],
-                                  failure: Throwable,
-                                  message: CallFailedRetryable,
-                                  data: WorkflowData,
-                                  events: Seq[ExecutionEventEntry]) = {
-    val currentSender = sender()
+  private def retryCall(callKey: ExecutionStoreKey,
+                        data: WorkflowData) = {
+
     val keyClone = callKey.retryClone
-    executionStore += keyClone -> ExecutionStatus.NotStarted
 
-    val completionWork = for {
-      _ <- globalDataAccess.setExecutionEvents(workflow.id, callKey.scope.fullyQualifiedName, callKey.index, callKey.attempt, events)
-      _ <- globalDataAccess.insertCalls(workflow.id, Seq(keyClone), backend)
-    } yield ()
+    val insertCopy = globalDataAccess.insertCalls(workflow.id, Seq(keyClone), backend)
 
-    completionWork onComplete {
-      case Failure(e) =>
-        logger.error(s"Preemption work failed for call ${callKey.tag}! " + e.getMessage, e)
-        currentSender ! CallActor.Ack(message)
+    insertCopy map { _ => PersistenceSucceeded(keyClone, ExecutionStatus.NotStarted) } pipeTo self
+
+    insertCopy onFailure {
```

Comment:
On the foreach I think the point is to avoid a callback (which is a good thing IMO), but you could still use onComplete as well. Either way you probably have a few years before it matters